apiary.apib

```diff
@@ -1661,9 +1661,261 @@ root will not be listable, as it is virtual. Access known paths directly.
 
 # Group name
 
-## publish
+IPFS namespace (IPNS) tool
 
-## resolve
+IPNS is a PKI namespace, where names are the hashes of public keys, and
+the private key enables publishing new (signed) values. In publish, the
+default value of <name> is your own identity public key.
+
+This command cannot be called directly.
+
+## publish [POST /name/publish{?arg}{&resolve,lifetime,ttl}]
+Publish an object to IPNS
+
++ Parameters
+    + arg (string, required) - IPFS path of the object to be published
+    + resolve (boolean, optional) - Resolve given path before publishing
+        + Default: `true`
+    + lifetime (string, optional) - Time duration that the record will be valid for. 
+        This accepts durations such as '300s', '1.5h' or '2h45m'. Valid time units are
+        'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'. Default: '24h'.
+    + ttl (string, optional) - Time duration this record should be cached for. Caution: experimental.
```

Comment:
What format does this have? The same as `lifetime`, if so would be good to have a custom type of `time` for these.